libraries/Microsoft.Bot.Connector/Authentication/ManagedIdentityServiceClientCredentialsFactory.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// A Managed Identity implementation of the <see cref="ServiceClientCredentialsFactory"/> interface.
+    /// </summary>
+    public class ManagedIdentityServiceClientCredentialsFactory : ServiceClientCredentialsFactory
```

Comment:
Is this exposed as public surface area for developers to use? or is this class only utilized within Microsoft.Bot.Builder.Integration.AspNet.Core ?  Can it be internal to that library?

Reply:
If developers do not use the default config based service client credentials factory, they could DI this class directly in startup.